akka-stream/src/main/scala/akka/stream/StreamUtils.scala

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (C) 2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.stream
+
+import akka.NotUsed
+import akka.actor.Cancellable
+import akka.stream.impl.fusing.GraphStages.SimpleLinearGraphStage
+import akka.stream.scaladsl.Flow
+import akka.stream.stage.{ GraphStageLogic, InHandler, OutHandler, StageLogging }
+import akka.util.OptionVal
+
+import scala.concurrent.duration.{ Duration, FiniteDuration }
+
+private[akka] object StreamUtils {
+  /**
+   * INTERNAL API
+   *
+   * Returns a flow that is almost identity but delays propagation of cancellation from downstream to upstream.
+   *
+   * Once the down stream is finish calls to onPush are ignored.
+   */
```

Comment:
`INTERNAL API` should go in the object scaladoc instead, and also the `@InternalApi` annotation on the object

Reply:
I moved it into Restart.scala as not to encourage reuse of stream.sleep :)